python/tests/test_transforms.py

```diff
@@ -506,3 +532,345 @@ def test_datetime_transform_str(transform, transform_str):
+        is None
+    )
+
+
+def test_projection_day_unary(bound_reference_timestamp) -> None:
+    assert DayTransform().project("name", BoundNotNull(term=bound_reference_timestamp)) == NotNull(term="name")
+
+
+def test_projection_day_literal(bound_reference_timestamp) -> None:
+    assert DayTransform().project(
+        "name", BoundEqualTo(term=bound_reference_timestamp, literal=TimestampLiteral(1667696874000))
+    ) == EqualTo(term="name", literal=literal(19))
+
+
+def test_projection_day_set_same_day(bound_reference_timestamp) -> None:
+    assert DayTransform().project(
+        "name",
+        BoundIn(term=bound_reference_timestamp, literals={TimestampLiteral(1667696874001), TimestampLiteral(1667696874000)}),
+    ) == EqualTo(term="name", literal=literal(19))
+
+
+def test_projection_day_set_in(bound_reference_timestamp) -> None:
+    assert DayTransform().project(
+        "name",
+        BoundIn(term=bound_reference_timestamp, literals={TimestampLiteral(1667696874001), TimestampLiteral(1567696874000)}),
+    ) == In(term="name", literals={literal(18), literal(19)})
+
+
+def test_projection_day_set_not_in(bound_reference_timestamp) -> None:
+    assert (
+        DayTransform().project(
+            "name",
+            BoundNotIn(term=bound_reference_timestamp, literals={TimestampLiteral(1567696874), TimestampLiteral(1667696874)}),
+        )
+        is None
+    )
+
+
+def test_projection_day_human(bound_reference_date: BoundReference) -> None:
+    date_literal = literal(date(2018, 1, 1))
+    assert DayTransform().project("dt", BoundEqualTo(term=bound_reference_date, literal=date_literal)) == EqualTo(
+        term="dt", literal=literal(17532)
+    )  # == 2018, 1, 1
+
+    assert DayTransform().project("dt", BoundLessThanOrEqual(term=bound_reference_date, literal=date_literal)) == LessThanOrEqual(
+        term="dt", literal=literal(17532)
+    )  # <= 2018, 1, 1
+
+    assert DayTransform().project("dt", BoundLessThan(term=bound_reference_date, literal=date_literal)) == LessThanOrEqual(
+        term="dt", literal=literal(17531)
+    )  # <= 2017, 12, 31
+
+    assert DayTransform().project(
+        "dt", BoundGreaterThanOrEqual(term=bound_reference_date, literal=date_literal)
+    ) == GreaterThanOrEqual(
+        term="dt", literal=literal(17532)
+    )  # >= 2018, 1, 1
+
+    assert DayTransform().project("dt", BoundGreaterThan(term=bound_reference_date, literal=date_literal)) == GreaterThanOrEqual(
+        term="dt", literal=literal(17533)
+    )  # >= 2018, 1, 2
+
+
+def test_projection_hour_unary(bound_reference_timestamp) -> None:
+    assert HourTransform().project("name", BoundNotNull(term=bound_reference_timestamp)) == NotNull(term="name")
+
+
+def test_projection_hour_literal(bound_reference_timestamp) -> None:
+    assert HourTransform().project(
+        "name", BoundEqualTo(term=bound_reference_timestamp, literal=TimestampLiteral(1667696874000))
```

Comment:
Minor: these tests use millisecond precision timestamps (Sun Nov 06 2022 01:07:54) but the actual representation is in microseconds. That's why the resulting hours ordinal is small. The actual date/time is Tue Jan 20 1970 07:14:56. It would be better to always use micros even when supplying constants so that future readers aren't confused.